### DISCUSSION TITLE
Automatic Achievement/Leaderboard Logic Archiving

### DISCUSSION THREAD

**Discussion by timenoe** [RetroAchievements/RAWeb#1036] [Ideas]
My request is the implementation of automatic logic archiving whenever a logic edit is made to an achievement or leaderboard. This would remove the need to backup old logic using something like pastebin and the data would be archived on the site, so we wouldn't have to rely on other sites for our historical data. In addition, the old logic will be able to be viewed in a readable way, as it is in the Dev drop-down.

Essentially, what I picture is a link in the Dev drop-down for an achievement that says "Version History" or something to that effect, which will show you the previous logic and the date range it was applicable. Every time the logic is updated, an archive would be sent here automatically.

Also, I don't know how feasible this is, but something like Vim Diff for logic would be incredible. In each archive, highlight lines that were added in green, and highlight lines that were removed from the previous version in red.

**Comment by Tsearo** [comment]
It's definitely one of the planned things we'd like to add once we have the database restructuring to handle it. 

Showing a Diff would be amazing but that's out of my knowledge on how difficult that would be to implement.

**Comment by Jamiras** [comment]
There's a more thorough discussion on ticket logic history in #1045, though it's focus is more on associating a specific historical version to a ticket.

**Comment by redwizard42** [comment]
Would this extend to backing up local logic as well, or would a local-file method be necessary for local achievement backups. I know when initially working on a set I've occasionally wished I could revert to an earlier version.

**Reply by Jamiras** [reply]
By definition, any server side versioning would only be for data sent to the server.

**Reply by redwizard42** [reply]
That's what I figured. At least that means my similar RAIntegration discussion isn't completely pointless.
